app/deed/server.py

```diff
@@ -99,4 +107,40 @@ def sign_deed(deed_, signature_):
         else:
             abort(status.HTTP_403_FORBIDDEN)
 
-        return jsonify(signature=signature), status.HTTP_200_OK
+        return {'signature': signature}, status.HTTP_200_OK
+
+    @blueprint.route('/deed/<deed_id>/completion', methods=['POST'])
+    def confirm_completion(deed_id):
+        def find_deed():
+            deed_ = Deed.get(deed_id)
+            if deed_ is None:
+                abort(status.HTTP_404_NOT_FOUND)
+            return deed_
+
+        def update_case_status():
+            response = case_api.update_status(deed.id, "Completion confirmed")
+            if response.status_code != status.HTTP_200_OK:
+                abort(status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        def update_deed():
+            try:
+                deed_json = copy.deepcopy(deed.json_doc)
```

Comment:
Is there any need to do a deep copies? just wondering why do this rather than suggesting you shouldn't. I suppose it's slower.